tests/components/august/test_init.py

```diff
@@ -170,3 +176,133 @@ async def test_set_up_from_yaml(hass):
         CONF_TIMEOUT: None,
         CONF_USERNAME: "mocked_username",
     }
+
+
+async def test_auth_fails(hass):
+    """Config entry state is ENTRY_STATE_SETUP_ERROR when auth fails."""
+
+    config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=_mock_get_config()[DOMAIN],
+        title="August august",
+    )
+    config_entry.add_to_hass(hass)
+    assert hass.config_entries.flow.async_progress() == []
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6184085)